### DISCUSSION TITLE
Plugin to synchronize the InvenTree database with the Mouser Electronics

### DISCUSSION THREAD

**Discussion by SergeoLacruz** [inventree/InvenTree#4334] [Show and tell]
Hi,
I wrote a scheduled task plugin that runs through the database, grabs infos from mouser and puts it into the database. 
Mainly price breaks. You can find it here:

https://github.com/SergeoLacruz/inventree-supplier-plugin

I added a bit of documentation. 
It runs on my setup since several days without problems :-). 
Perhaps it is useful for other people too. Suggestions welcome. Have a nice weekend,

Michael

**Comment by matmair** [comment]
Very nice! I am actually working on a generic solution to enable things like this built-in with #3791 .
Would it be ok if I add this to the plugin repo?

**Comment by SergeoLacruz** [comment]
A generic solution would be even better.
You can add the plugin to the repo if you think it is good enough. Its partly special to ours setup and might need modification for other setups.

**Comment by eeintech** [comment]
@SergeoLacruz Very nice! Except one thing would prevent us to use it is the automatic supplier part creation. We buy most of our parts on Digi-Key and have a few supplier parts linked to Mouser, I don't want to clutter our database with new unnecessary supplier parts. If you add the option to NOT create supplier part automatically then I'll give your plugin a shot. Thanks!

**Comment by SergeoLacruz** [comment]
Hi,
As I already wrote in the description, this feature is questionable. It can be taken out easily, either using a switch as you suggested or just deleted. Anyhow if you are working with Digikey, the plugin will no help you becaue it does not use Digikey as a source. I planned to expand it but it might make senst to wait for #3791 .

**Reply by eeintech** [reply]
I read it but I am proposing to implement it for people who still want to update Mouser pricing only for parts with Mouser entries. I think this is a must if you want people to try it out without messing-up their database.

**Comment by SergeoLacruz** [comment]
Done :-)
But keep in mind, if you have a large database and only a few Mouser parts it does not make much sense to use it that way because the part is just skipped and the slot it gone. So the plugin will not be faster then before. It would make more sense to modify the plugin to loop over SuppllierParts instead of Parts. May be some day...

**Reply by eeintech** [reply]
Oh I see... what is the reason you're looping over Parts instead of SupplierParts? One could think such supplier synchronization plugin shouldn't manipulate Parts?

**Comment by SergeoLacruz** [comment]
We just want to have it like that :-) Changing the way it works isn't a big deal. Its is just the main function. All the rest can stay.

**Comment by SergeoLacruz** [comment]
After more than a year I came back to this. The automatic creation of supplier parts is gone :-)
Instead the plugin creates a list of changes and and just lists new parts available at Mouser. From the list the user can 
individually decide what to do with this. To store that list, the plugin makes use of the AppMixin that's marked dangerous. 
As this is my first experience with this mixin, anyone more skilled in the art might give a hint if something is bad.
